### DISCUSSION TITLE
How can I get the shear stress term of x?

### DISCUSSION THREAD

**Discussion by seongwonHam** [idaholab/moose#23577] [Q&A Modules: Solid mechanics] [answered]
Hello, MOOSE developers.

Now I'm trying to make the basic code to certain the compressive and shear stress in the 3D cubic mesh.
My code is like below;

```
[Mesh]
  type = FileMesh
  file = 3D.e
[]

[GlobalParams]
  displacements = 'disp_x disp_y disp_z'
[]

[Modules/TensorMechanics/Master]
  [./block1]
    strain = FINITE
    add_variables = true
    generate_output = 'stress_xx stress_yy stress_zz stress_xy stress_yz stress_xz'
  [../]
[]

[Materials]
  [./elasticity_tensor]
    type = ComputeElasticityTensor
    C_ijkl = '59.99e9 34.30e9 26.70e9 30.62e9 4.09e9'
    fill_method = axisymmetric_rz
    block = 1
  [../]
  [./stress]
    type = ComputeFiniteStrainElasticStress
  [../]
[]

[BCs]
  [./fixed_top]
    type = DirichletBC
    variable = disp_z
    boundary = z_plus
    value = 0
  [../]
  [./fixed_left]
    type = DirichletBC
    variable = disp_x
    boundary = 'x_minus x_plus'
    value = 0
  [../]
  [./fixed_right]
    type = DirichletBC
    variable = disp_y
    boundary = 'y_minus y_plus'
    value = 0
  [../]
  [./Pressure]
    [./comp]
      boundary = z_minus
      function = 1e8*t
    [../]
  [../]
[]

[Preconditioning]
  [SMP]
    type = SMP
    full = true
  []
[]

[Executioner]
  type = Transient
  solve_type = 'PJFNK'
  start_time = 0
  n_startup_steps = 1
  end_time = 1
  dt = 0.01
[]

[Outputs]
  exodus = true
[]
```

and my mesh file is attached.  
![3D](https://user-images.githubusercontent.com/106044260/222344151-3008d2e2-f4e4-49c6-945a-8aed2e5b48ce.gif)
please change the name of this file to 3D.e (exodus file).

When I use the above input file and exodus file to run the MOOSE calculation, I can get the result file.
But it doesn't contain "stress_xx stress_xy stress_xz" even though I write those outputs in my input file.
![image](https://user-images.githubusercontent.com/106044260/222344563-945486ec-5a7e-42c3-a8da-4c2e042aab7e.png)

I have checked it using Paraview, but output variables seem like this picture.
Please let me know how can I fix this code.

**Comment by seongwonHam** [comment]
https://drive.google.com/file/d/1wBiWDIwZL7YFP3eaEk6UZjNFnums9KS-/view?usp=sharing

I share my exodus mesh file using google drive.

**Comment by Rahim-Habibi** [comment, marked as answer]
Hi, 
You are right, you mentioned "stress_xx stress_xy stress_xz" in input file and they were calculated in your results. The point is all of "stress_xx stress_xy stress_xz" components are being saved as a single vector, you can check them in Paraview while other components like "stress_yz are being saved as scalar.

**Reply by GiudGiud** [reply]
Yeah basically Paraview is grouping stress_xy, stress_xx and stress_xz under stress_x

**Reply by seongwonHam** [reply]
I appreciate your kind reply!

Then, How can I see the "stress_x" vector as separated scalar terms "stress_xx stress_xy stress_xz" in Paraview?

**Reply by GiudGiud** [reply]
You could add a suffix in the material property output
click on it and select each component

**Reply by seongwonHam** [reply]
Thanks a lot! I will try it.
